realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/utils/UserFactory.java

```diff
@@ -32,4 +56,43 @@ public static SyncUser createAdminUser(String authUrl) {
         SyncCredentials credentials = SyncCredentials.custom("admin", "debug", null);
         return SyncUser.login(credentials, authUrl);
     }
+
+    // Since we don't have a reliable way to reset the sync server and client, just use a new user for every test case.
+    public static void resetInstance() {
```

Comment:
There seems to be a mismatch between the name and description. Perhaps `generateNewUser()` instead?